Microsoft/WebTemplateStudio
# Dependency checker for frameworks page

**How it works**
- When the Select Frameworks page loads, the component uses `vscode.postMessage` to tell the extension to figure out the dependency installation state
- Extension receives the command, calls `checkDependency` function in dependencyChecker.ts
- Extension returns a payload with the installation state to the client
- Client listens and receives the payload in `App.tsx` in the `case EXTENSION_COMMANDS.GET_DEPENDENCY_INFO` and dispatches an action to update the Redux store with the dependency info
- `SelectableCard` uses the information from the Redux store and renders a `DependencyInfo` component

**Client side changes**
- App.tsx: add a new listener that listens for the payload from the extension
- Added new type, action and reducer with WIZARD_INFO_TYPEKEYS.UPDATE_DEPENDENCY_INFO
- Yellow warning svg
- DependencyInfo component: takes in framework name and installation state and displays a warning (this is gated with a preview flag)
- SelectFrameworks is moved into containers folder from components since I use the Redux store, namely the vscode API to send payloads to the extension when this component is loaded (frameworks page)
- New prop `isFrameworkSelection` passed down to `SelectableCards` because this component is actually shared between all the pages, I only want the dependency info showing on the frameworks page
- 2 new strings in translations file
- mockVsCodeApi - this is used when the client is run in the browser and mimics communication with the extension

**Extension side changes**
- src/constants.ts: Added new command and extension module for dependency checker
- src/controller.ts: Added dependency checker to the extensionModuleMap
- src/utils/dependencyChecker.ts: logic for checking dependency (we are currently checking for node and python3.x, the python command depends on users' operating system)

**How to test**
- Build the project
- Go to VS Code setting and enable preview mode
- Launch the extension and go to the Framework page
- Try installing/uninstalling Python/Node
- Go to another page, and come back to the Framework page, and the dependency checker status should change
- DependencyInfo should not show up if preview mode is not enabled
